event/datadeal/data_verification_event.go

```diff
@@ -1 +1,231 @@
+
+	dealID, err := strconv.ParseUint(dealIDStr, 10, 64)
+	if err != nil {
+		return err
+	}
+
+	voteOption := d.verifyAndGetVoteOption(dealID, dataHash)
+
+	msgVoteDataVerification, err := makeDataVerificationVote(
+		d.reactor.OracleAcc().GetAddress(),
+		dataHash,
+		dealID,
+		voteOption,
+		d.reactor.OraclePrivKey().Serialize(),
+	)
+	if err != nil {
+		return err
+	}
+
+	txBuilder := panacea.NewTxBuilder(*d.reactor.QueryClient())
+
+	txBytes, err := generateTxBytes(msgVoteDataVerification, d.reactor.OracleAcc().GetPrivKey(), d.reactor.Config(), txBuilder)
+	if err != nil {
+		return fmt.Errorf("generate tx failed. dealID(%d). dataHash(%s): %v", dealID, dataHash, err)
+	}
+
+	if err := broadcastTx(d.reactor.GRPCClient(), txBytes); err != nil {
+		return fmt.Errorf("broadcast transaction failed. dealID(%d). dataHash(%s): %v", dealID, dataHash, err)
+	}
+
+	return nil
+}
+
+func (d DataVerificationEvent) decryptData(decryptedSharedKey, nonce, encryptedDataBz []byte) ([]byte, error) {
+	decryptedData, err := crypto.DecryptWithAES256(decryptedSharedKey, nonce, encryptedDataBz)
+	if err != nil {
+		return nil, err
+	}
+	return decryptedData, nil
+}
+
+func (d DataVerificationEvent) compareDataHash(dataSale *types.DataSale, decryptedData []byte) bool {
+	decryptedDataHash := sha256.Sum256(decryptedData)
+	decryptedDataHashStr := hex.EncodeToString(decryptedDataHash[:])
+
+	return decryptedDataHashStr == dataSale.DataHash
+}
+
+func (d DataVerificationEvent) convertSellerData(deal *types.Deal, dataSale *types.DataSale) ([]byte, error) {
+	encryptedDataBz, err := d.reactor.Ipfs().Get(dataSale.VerifiableCid)
+	if err != nil {
+		return nil, err
+	}
+
+	oraclePrivKey := d.reactor.OraclePrivKey()
+
+	sellerAcc, err := d.reactor.QueryClient().GetAccount(dataSale.SellerAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	sellerPubKeyBytes := sellerAcc.GetPubKey().Bytes()
+
+	sellerPubKey, err := btcec.ParsePubKey(sellerPubKeyBytes, btcec.S256())
+	if err != nil {
+		return nil, err
+	}
+
+	decryptSharedKey := crypto.DeriveSharedKey(oraclePrivKey, sellerPubKey, crypto.KDFSHA256)
+
+	decryptedData, err := d.decryptData(decryptSharedKey, deal.Nonce, encryptedDataBz)
+	if err != nil {
+		return nil, err
+	}
+
+	return decryptedData, nil
+}
+
+func (d DataVerificationEvent) verifyAndGetVoteOption(dealID uint64, dataHash string) oracletypes.VoteOption {
+	deal, err := d.reactor.QueryClient().GetDeal(dealID)
+	if err != nil {
+		if errors.Is(err, types.ErrDataSaleNotFound) {
+			log.Infof("failed to find deal (%d)", dealID)
+			return oracletypes.VOTE_OPTION_NO
+		} else {
+			return oracletypes.VOTE_OPTION_UNSPECIFIED
+		}
+	}
+
+	dataSale, err := d.reactor.QueryClient().GetDataSale(dataHash, dealID)
+	if err != nil {
+		if errors.Is(err, types.ErrDataSaleNotFound) {
+			log.Infof("failed to find dataSale (%s)", dataHash)
+			return oracletypes.VOTE_OPTION_NO
+		} else {
+			return oracletypes.VOTE_OPTION_UNSPECIFIED
+		}
+	}
```

Comment:
`ErrDataSaleNotFound` means no data sale for vote ([as gyuguen said](https://github.com/medibloc/panacea-doracle/pull/91#discussion_r1009156876)).
That is vote tx is meaningless, so we can return `oracletypes.VOTE_OPTION_UNSPECIFIED` for all errors in query of data sale.

Reply:
Fixed in [8ee1242](https://github.com/medibloc/panacea-doracle/pull/89/commits/8ee1242b70252bc579b95b0e16ccbe204ba77e01).